lib/hyrax/acl/change_set.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+module Hyrax
+  module Acl
+    ##
+    # @api private
+    #
+    # Build a changeset class for the given resource class. The ChangeSet will
+    # have fields to match the resource class given.
+    #
+    # To define a custom changeset with validations, use naming convention with "ChangeSet" appended to the end
+    # of the resource class name. (e.g. for BookResource, name the change set BookResourceChangeSet)
+    #
+    # @example
+    #   Hyrax::ChangeSet(Monograph)
+    def self.ChangeSet(resource_class)
+      klass = (resource_class.to_s + "ChangeSet").safe_constantize || Hyrax::Acl::ChangeSet
```

Comment:
It feels a little bit weird to be adding these complex change set behaviours to what is otherwise a very minimal gem but I understand if you feel they are necessary. (Are they? The `ChangeSet` used by `AccessControlList` isn’t ever exposed, so is there a problem with it being more minimal? Or is subclassing `AccessControlList` something we’re worried about?)

Reply:
it seemed like it would be more work to undo this than to simply port it in as-is.

i think it would be fine to provide static change sets each for Permission and AccessControl (or even to skip Permission, since we don't use it internally), but i'd rather not block feature parity on a rewrite of this part of the codebase.